### PR TITLE
Set gridline (if available) as the default grid registration for remote datasets

### LINKED ISSUE #1929: load_earth_relief now returns grid-registrated grids when registration is not specified
**Description of the problem**

In GMT<=6.3, when users access a remote grid but don't specify the grid registration (e.g., `@earth_relief_01d`), the pixel-registrated grid is returned by default unless only the gridline-registrated grid is available.

The behavior has changed recently (https://github.com/GenericMappingTools/gmt/pull/6710). The new behavior is well documented in https://docs.generic-mapping-tools.org/dev/datasets/remote-data.html#data-registration.

> Optionally, you can append _g or _p to specifically get the gridline-registered or pixel-registered version (if they both exist). If reg is not specified then the behavior depends on whether you are making a plot or processing/extracting a subset of the data:
> 
> 1. For plots we will return the pixel-registered version unless only the gridline-registered file is available.
> 2. For grid processing modules we will return the gridline-registered version unless only the pixel-registered file is available. We will also issue a warning since for calculations you should ideally know and specify exactly what you want.
> 
> If you do specify a specific registration and that version is not available you will get an error message.

In PyGMT, we provide the `load_earth_relief` function to load the earth relief dataset into a xarray.DataArray object. When `registration` is not specified, the function returns a pixel-registrated grid in GMT<=6.3, but will return a gridline-registrated grid in GMT 6.4, thus a few tests will fail.

### PR DESCRIPTION
**Pull request by willschlitzer**
This sets the default registration to `"gridline"` if both registration options are available.

Fixes #1929

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/load_remote_dataset.py
Comment (willschlitzer) [reply pygmt/datasets/load_remote_dataset.py:217]: Okay. I will be back on December 26 and came update the PR then.

**Comment by willschlitzer** [comment]
Should I add tests using high-resolution grids to make sure the functions default to the appropriate registrations when only one is available?

**Comment by seisman** [comment]
> Should I add tests using high-resolution grids to test that the functions default to the appropriate registrations when only one is available?

Yes, please.

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [reply pygmt/datasets/load_remote_dataset.py:217]: > In either case, I also think we should put `gridline` before `pixel`, because letter `g` is in front of letter `p` and `gridline` and `pixel` are internally represented as 0 and 1 in GMT.

I feel these changes should be done in a separate PR so that this PR is smaller and easier to review.

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:90]:
```suggestion
```

**Comment by willschlitzer** [comment]
@GenericMappingTools/pygmt-maintainers I wrote the high-resolution tests looking for default registrations to also duplicate the tests run in the other high-resolution region tests (e.g. `test_earth_faa_05m_with_region()`). To save cut down on testing, should I remove the `05m` tests and the associated cache files?

**Comment by seisman** [comment]
> should I remove the `05m` tests and the associated cache files?

Sounds good to me.

File: pygmt/datasets/earth_age.py
Comment (weiji14) [inline pygmt/datasets/earth_age.py:48]: Looking at https://www.generic-mapping-tools.org/remote-datasets/earth-age.html, it seems like gridline `is` the default for all resolutions anyway? Not sure what the behaviour was for GMT 6.3 though.

```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``.
```

File: pygmt/datasets/earth_free_air_anomaly.py
Comment (weiji14) [inline pygmt/datasets/earth_free_air_anomaly.py:48]: What about rewording it explicitly to something like this?

```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``
        for all resolutions except ``"01m"`` which is ``"pixel"`` only.
```

Comment (weiji14) [inline pygmt/datasets/earth_free_air_anomaly.py:48]: What about rewording it explicitly to something like this? Ref: https://www.generic-mapping-tools.org/remote-datasets/earth-faa.html

```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``
        for all resolutions except ``"01m"`` which is ``"pixel"`` only.
```

File: pygmt/datasets/earth_geoid.py
Comment (weiji14) [inline pygmt/datasets/earth_geoid.py:48]: Ref: https://www.generic-mapping-tools.org/remote-datasets/earth-geoid.html
```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``.
```

File: pygmt/datasets/earth_magnetic_anomaly.py
Comment (weiji14) [inline pygmt/datasets/earth_magnetic_anomaly.py:53]: Ref: https://www.generic-mapping-tools.org/remote-datasets/earth-mag.html

```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``
        for all resolutions except ``"02m"`` which is ``"pixel"`` only.
```

File: pygmt/tests/test_datasets_earth_age.py
Comment (seisman) [inline pygmt/tests/test_datasets_earth_age.py:62]: Please also update `test_earth_age_05m_without_region` to use the `01m` resolution, so that we consistently use `01d` and `01m` in the tests. 

Please also make the similar changes in other tests.

File: pygmt/datasets/earth_vertical_gravity_gradient.py
Comment (weiji14) [inline pygmt/datasets/earth_vertical_gravity_gradient.py:50]: Ref: https://www.generic-mapping-tools.org/remote-datasets/earth-vgg.html
```suggestion
        ``"gridline"`` for gridline registration. Default is ``"gridline"``
        for all resolutions except ``"01m"`` which is ``"pixel"`` only.
```

**Review by weiji14 — COMMENTED** [review]
Prefer explicit documentation on what resolutions default to `gridline` and which are `pixel` only. Feel free to change the wording if needed.

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (seisman) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:130]: `test_earth_mag4km_05m_with_region():` is still using the `05m` grid.

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (seisman) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:66]: `earth_mag` doesn't have `01m` resolution.
```suggestion
        load_earth_magnetic_anomaly("02m")
```

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (seisman) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:60]:
```suggestion
def test_earth_mag_02m_without_region():
```

**Comment by willschlitzer** [comment]
@seisman and @weiji14 Please feel free to commit any minor changes you suggest to this PR and #2241. I'm going to be away from my computer for a good portion of today, and don't want to hold up @yvonnefroehlich over something minor.

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
> @seisman and @weiji14 Please feel free to commit any minor changes you suggest to this PR and #2241. I'm going to be away from my computer for a good portion of today, and don't want to hold up @yvonnefroehlich over something minor.

I've made the commit cedfa26efebc8792c488f81883f62bcc9741e4ce to make sure the 01d resolution always returns gridline-registered grids.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**
